components/sup/src/sup.c

```diff
@@ -8,13 +9,131 @@
 #include <opencan_templates.h>
 #include <opencan_tx.h>
 
+typedef enum {
+	SEG_A  = 13,
+	SEG_B  = 4,
+	SEG_C  = 7,
+	SEG_D  = 8,
+	SEG_E  = 9,
+	SEG_F  = 12,
+	SEG_G  = 5,
+	SEG_DP = 11,
+	SEG_1  = 1,
+	SEG_2  = 2,
+	SEG_3  = 3,
+	SEG_4  = 6,
+} SEVEN_SEG_PINS;
+
+static void bts_authorization();
+static void init_led();
 static void sup_100Hz();
+static void init_pin(SEVEN_SEG_PINS pin);
+static void set_one();
+static void set_zero();
+
 
 static bool bbc_authorized;
 static bool throttle_authorized;
 static bool steer_authorized;
 
+
+static const SEVEN_SEG_PINS zero[]
+	= {SEG_A, SEG_B, SEG_C, SEG_D, SEG_E, SEG_F};
+
+static const SEVEN_SEG_PINS one[] = {SEG_B, SEG_C};
+
+static enum {
+	bbc_state,
+	throttle_state,
+	steer_state
+} current_state = bbc_state;
+
+static void init_pin(SEVEN_SEG_PINS pin)
```

Comment:
I see the vision, but this is common use-case, so look into using [`gpio_config_t`](https://docs.espressif.com/projects/esp-idf/en/v5.4.1/esp32s3/api-reference/peripherals/gpio.html#configure-usb-phy-pins-to-gpio) to achieve similar behavior.